c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,117 @@
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        private bool _isAbsoluteDiff;
+        private bool _isDiffIndented;
+
+        public JsonDiff(bool isAbsoluteDiff, bool isDiffIndented)
+        {
+            _isAbsoluteDiff = isAbsoluteDiff;
+            _isDiffIndented = isDiffIndented;
+        }
+
+        public JsonDiff()
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            _isAbsoluteDiff = true;
+            _isDiffIndented = true;
+        }
+
+        public JObject NumericDiff(string left, string right)
+        {
+            string diff = "";
+            try
             {
-                if (firstKeys.Contains(key))
+                diff = JsonDiffBetweenTwo(left, right);
+                if (diff == null)
                 {
-                    matches.Add(key);
+                    return JObject.Parse(@"{}");
                 }
+                JObject diffJsonObject = JObject.Parse(diff);
+                JObject diffReturnObject = CalculateNumericDiff(diffJsonObject, _isAbsoluteDiff);
+                return diffReturnObject;
             }
-            foreach(string key in matches)
+            catch (Exception e)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                return JObject.Parse(diff + e.StackTrace);
+            }
+        }
+
+        public string NumericDiffAsString(string left, string right)
+        {
+            JObject diff = NumericDiff(left, right);
+            return _isDiffIndented ? diff.ToString(Formatting.Indented) : diff.ToString();
+
+        }
+
+        private JObject CalculateNumericDiff(JObject jObject, bool isAbsoluteDiff)
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
+            {
+                if (prop.Key == "_t")
+                {
+                    continue;
+                }
+                if (prop.Value != null)
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    if (prop.Value is JArray)
+                    {
+                        JArray valuesAsArray = (JArray)prop.Value;
```

Comment:
You can consolidate the type checking and assignment into the `if` statement. Something like `if (prop.Value is JArray valuesAsArray)` - then you don't need to explicitly cast it again.